pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -359,7 +359,7 @@ public CompletableFuture<Void> initialize() {
                     this.isEncryptionRequired = policies.encryption_required;
 
                     isAllowAutoUpdateSchema = policies.is_allow_auto_update_schema;
-                })
+                }, brokerService.getTopicOrderedExecutor().chooseThread(topic))
```

Comment:
it'd be good to create a private method to get an executor instead of repeating this call many times

Reply:
in the Dispatcher we cached the result of brokerService.getTopicOrderedExecutor().chooseThread(topic) into an  instance field, this will also reduce the waste of resources due to choosing the thread (and make the code cleaner)

Reply:
thanks for the suggestion, I'll do that.